codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -728,7 +728,7 @@ private class ServerHttpProtocolImplGenerator(
                         rustTemplate(
                             """
                             input = input.${binding.member.setterName()}(
-                                #{deserializer}(m$index)?
+                                ${symbolProvider.toOptional(binding.member, "#{deserializer}(m$index)?")}
```

Comment:
maybe we should just make #{deserializer} always return `Option<T>`?

Reply:
I think I'd prefer to keep the `#{deserializer}` as it is and handle this in the caller here TBH.. It looks logically more correct to not return always an `Option<T>` from it if it is not needed..